reporting-app/db/migrate/20251210000001_create_ex_parte_activities.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+class CreateExParteActivities < ActiveRecord::Migration[7.2]
+  def change
+    create_table :ex_parte_activities, id: :uuid, comment: "Hours data from external sources (API/batch) for compliance calculation" do |t|
+      t.string :member_id, null: false,
+               comment: "Member reference - always required (allows hours-first save order)"
+      t.uuid :certification_id,
+             comment: "Certification reference - nullable for pending entries before cert creation"
+      t.string :category, null: false,
+               comment: "Activity category: employment, community_service, education"
+      t.decimal :hours, precision: 8, scale: 2, null: false,
+                comment: "Hours worked/volunteered (max 744 = 31 days × 24 hours)"
+      t.date :period_start, null: false,
+             comment: "Activity period start date"
+      t.date :period_end, null: false,
+             comment: "Activity period end date"
+      t.boolean :outside_period, default: false, null: false,
+                comment: "True if activity dates are outside certification lookback period"
+      t.string :source_type, null: false,
+               comment: "Source type: 'api' or 'batch_upload'"
+      t.uuid :source_id,
+             comment: "Source record ID (e.g., HoursDataBatchUpload ID for batch_upload)"
+      t.datetime :reported_at, null: false,
```

Comment:
question: is this like when the data was originally obtained by the state in their system?

Reply:
yes this was intended for that , again I have seen this needed in my experience specially when troubleshooting and reporting.

Reply:
but I will remove for now since it speculative at this time

Reply:
got it thanks for explaining. yeah i can see that this is would be valuable.